assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,143 @@
+import { Fragment, useEffect, useState, useCallback } from "@wordpress/element";
+import { PanelBody } from '@wordpress/components';
+import { MediaPlaceholder } from "@wordpress/editor";
+import { InspectorControls } from '@wordpress/block-editor';
+
+import { URLInput } from "../../components/URLInput/URLInput";
+import { MediaEmbedPreview } from "./MediaEmbedPreview";
+import { MediaElementVideo } from './MediaElementVideo';
+import { useSelect } from '@wordpress/data';
+import { debounce } from 'lodash';
+
+const { __ } = wp.i18n;
+const { RichText } = wp.blockEditor;
+
+const MediaInspectorOptions = ({ attributes, setAttributes }) => {
+  // Using a state to prevent the input losing the cursor position, a React issue reported multiple times
+  const [ mediaURL, setMediaURL ] = useState(attributes.youtube_id);
+  const debouncedMediaURLUpdate = useCallback(debounce(value => setAttributes({ youtube_id: value }), 300), []);
```

Comment:
Ah, the debounce is there to prevent making a request to the oEmbed API on every keystroke, it's not related to the cursor position issue.

Reply:
Ah ok :+1: . My suggestion still holds as it seems a simpler way to achieve the same thing. The debounce would be bit simpler too then:
```
        <URLInput
          value={ mediaURL }
          onChange={value => debounce(() => setMediaURL(value), 300) }
         // ...
        />
```

Reply:
I had to defer to `useCallback` because of this issue: 
https://stackoverflow.com/questions/55616536/lodash-debounce-in-react-functional-component-not-working
https://dev.to/pulkitnagpal/using-throttling-and-debouncing-with-react-hooks-57f1

I tested your snippet anyway just in case I was wrong but it actually prevents you from typing anything in the input field. 

At this point I would avoid looking for more elegant ways of doing something that's working with no issues to prevent any further delays, it's hard to determine which comments are blockers and which aren't. Maybe we should use a "traffic light" system (red, yellow, green) to flag these.

Note: this block is probably going to be deprecated and replaced by the native WP blocks in the near future (cc @comzeradd )

Reply:
But what would be the benefit of adding another hook? It may look simpler in the `onChange` call, but why adding an effect when there is no real need for an effect there? There is [a comment](https://stackoverflow.com/questions/55616536/lodash-debounce-in-react-functional-component-not-working/58929988#comment109952426_58929988) to that solution actually suggesting this.

Reply:
Yeah maybe that would only make sense if there were multiple things calling `setMediaURL`, then we would only have to add the debounce in one place. But since that won't be the case here, feel free to ignore these comments :+1: